File: sdk/php/src/Model/TrajectoryPoint.php
<?php

namespace HJTPX\Captcha\Model;

class TrajectoryPoint
{
    public $x;
    public $y;
    public $t;

    public function __construct(int $x, int $y, int $t)
    {
        $this->x = $x;
        $this->y = $y;
        $this->t = $t;
    }

    public static function fromArray(array $data): self
    {
        return new self($data['x'], $data['y'], $data['t']);
    }

    public function toArray(): array
    {
        return [
            'x' => $this->x,
            'y' => $this->y,
            't' => $this->t,
        ];
    }
}

File: sdk/java/src/main/java/com/hjtpx/captcha/model/TrajectoryPoint.java
package com.hjtpx.captcha.model;

import com.fasterxml.jackson.annotation.JsonProperty;

public class TrajectoryPoint {
    @JsonProperty("x")
    private int x;

    @JsonProperty("y")
    private int y;

    @JsonProperty("t")
    private long timestamp;

    public TrajectoryPoint() {
    }

    public TrajectoryPoint(int x, int y, long timestamp) {
        this.x = x;
        this.y = y;
        this.timestamp = timestamp;
    }

    public int getX() {
        return x;
    }

    public void setX(int x) {
        this.x = x;
    }

    public int getY() {
        return y;
    }

    public void setY(int y) {
        this.y = y;
    }

    public long getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(long timestamp) {
        this.timestamp = timestamp;
    }
}
